### ISSUE TITLE
[question] Possible to open commit in terminal buffer?

### ISSUE THREAD

**Issue by jlesquembre** [rbong/vim-flog#8]
First of all, thanks for writing vim-flog, I just started to use it and I find it very useful.

I have a question, in the `Flog` window, `<CR>` opens the commit, but would be possible to open it in a terminal buffer? I'm asking because I'd like to have the commit formatted with [diff-so-fancy](https://github.com/so-fancy/diff-so-fancy)

**Comment by rbong** [comment]
Please note that you'll lose integration with fugitive if you do this - no filetype settings for commits, no navigation bindings (`<CR>` over a filename to open that file or over a diff chunk to open that chunk), no running fugitive commands from the buffer - so this will probably never be fully supported since fugitive integration is an integral part of the plugin.

Something like this will work though:

```vim
augroup Myfloggroup
    autocmd FileType floggraph command! Myflogsplitcommit call flog#preview_commit('vertical belowright term ' . flog#get_fugitive_git_command() . ' -p show ', 1)
    autocmd FileType floggraph nnoremap <buffer> <silent> <CR> :Myflogsplitcommit<CR>
augroup END
```

I haven't tested this with diff-so-fancy but the `-p` option should force git to use the pager.

Also, I've passed in `1` to `preview_commit` so it keeps focus because you'll have to close the window manually while the pager is running.

**Comment by rbong** [comment]
If there are any problems with the answer please feel free to reopen.

**Comment by jlesquembre** [comment]
It's fine if I lose fugitive integration in this case, since 90% of the time, I just want to see the diff, and for me it's more important a nicer visualization.

Thanks for the snippet, it works, but it opens the terminal in the current window. To open a new terminal in a split window and run a command there, I usually do this:
```vim
:vnew | call termopen('git diff')
```

It took me some time, but I found a way to open a new terminal window and show the diff with diff-so-fancy:
https://github.com/jlesquembre/dotfiles/blob/0c887243ebb24fa8a58902555903a076ce4adf5a/nvim/autoload/myflog.vim#L22-L39

There are some other helper functions on that file.

In case anyone wants to have a setup similar to mine, I also added this maps:
https://github.com/jlesquembre/dotfiles/blob/0c887243ebb24fa8a58902555903a076ce4adf5a/nvim/init.vim#L945-L954

`<CR>` opens the fancy diff and `o` opens fugitive diff, only one diff is allowed at any time. 
`<c-n>` / `<c-p>` goes to the next / previous commit using the current opened diff
`J` / `K` scrolls the fancy diff from the flog window

I'll try to add a short screencast if I find some time. 

In case you find the fancy diff preview interesting, I can try to polish it and make a PR. If not, it's totally fine with me. Thanks again for flog, it's really useful.
